python/cuml/cuml/tests/test_umap.py

```diff
@@ -830,30 +831,52 @@ def test_umap_distance_metrics_fit_transform_trust_on_sparse_input(
 @pytest.mark.parametrize("num_clusters", [0, 3, 5])
 @pytest.mark.parametrize("fit_then_transform", [False, True])
 @pytest.mark.parametrize("metric", ["l2", "sqeuclidean", "cosine"])
+@pytest.mark.parametrize("do_snmg", [True, False])
 def test_umap_trustworthiness_on_batch_nnd(
-    data_on_host, num_clusters, fit_then_transform, metric
+    data_on_host, num_clusters, fit_then_transform, metric, do_snmg
 ):
 
     digits = datasets.load_digits()
 
+    umap_handle = None
+    if do_snmg:
+        umap_handle = DeviceResourcesSNMG()
+
     cuml_model = cuUMAP(
+        handle=umap_handle,
         n_neighbors=10,
         min_dist=0.01,
         build_algo="nn_descent",
-        build_kwds={"nnd_n_clusters": num_clusters},
+        build_kwds={"n_clusters": num_clusters},
         metric=metric,
     )
 
-    if fit_then_transform:
-        cuml_model.fit(
-            digits.data, convert_dtype=True, data_on_host=data_on_host
-        )
-        cuml_embedding = cuml_model.transform(digits.data)
-    else:
-        cuml_embedding = cuml_model.fit_transform(
-            digits.data, convert_dtype=True, data_on_host=data_on_host
-        )
+    def run_umap():
+        if fit_then_transform:
+            cuml_model.fit(
+                digits.data, convert_dtype=True, data_on_host=data_on_host
+            )
+            cuml_embedding = cuml_model.transform(digits.data)
+        else:
+            cuml_embedding = cuml_model.fit_transform(
+                digits.data, convert_dtype=True, data_on_host=data_on_host
+            )
+
+        return cuml_embedding
+
+    # num clusters should be >= 1
+    if num_clusters == 0:
+        with pytest.raises(ValueError):
+            run_umap()
+        return
+
+    # data should be on host if batching (num_clusters > 1)
+    if num_clusters > 1 and not data_on_host:
+        with pytest.raises(Exception):
```

Comment:
Should this be
```suggestion
        with pytest.raises(ValueError):
```

Reply:
This has to be an Exception because it is thrown from cuVS, and not at the python level.